api/v2/openapi.yaml

```diff
@@ -0,0 +1,345 @@
+          default: true
+        - in: query
+          name: inhibited
+          type: boolean
+          description: Show inhibited alerts
+          default: true
+        - in: query
+          name: unprocessed
+          type: boolean
+          description: Show unprocessed alerts
+          default: true
+        - name: filter
+          in: query
+          description: A list of matchers to filter alerts by
+          required: false
+          type: array
+          items:
+            type: string
+        - name: receiver
+          in: query
+          description: A regex matching receivers to filter alerts by
+          required: false
+          type: string
+      responses:
+        '200':
+          description: Get alerts response
+          schema:
+            type: array
+            items:
+              '$ref': '#/definitions/alert'
+        '400':
+          description: Bad request
+          schema:
+            type: string
+        '500':
+          description: Internal server error
+          schema:
+            type: string
+    post:
+      tags:
+        - alert
+      operationId: postAlerts
+      description: Create a new Alert
+      consumes:
+        - application/json
+      parameters:
+        - in: body
+          name: alerts
+          description: the alerts to create
+          required: true
+          schema:
+            $ref: '#/definitions/alerts'
+      responses:
+        '200':
+          description: Get alerts response
+        '500':
+          description: Internal server error
+          schema:
+            type: string
+        '400':
+          description: Bad request
+          schema:
+            type: string
+
+definitions:
+  alertmanagerStatus:
+    type: object
+    properties:
+      name:
+        type: string
+      statusInCluster:
+        type: string
+      peers:
+        type: array
+        minimum: 0
+        items:
+          $ref: '#/definitions/peerStatus'
+    required:
+      - name
+      - statusInCluster
+      - peers
+  peerStatus:
+    type: object
+    properties:
+      name:
+        type: string
+      address:
+        type: string
+  silence:
+    type: object
+    properties:
+      id:
+        type: string
+      matchers:
+        $ref: '#/definitions/matchers'
+      startsAt:
+        type: string
+        format: date-time
+      endsAt:
+        type: string
+        format: date-time
+      updatedAt:
+        type: string
+        format: date-time
+      createdBy:
+        type: string
+      comment:
+        type: string
+      status:
+        type: object
+        properties:
+          state:
+            type: string
+            enum: ["expired", "active", "pending"]
+    required:
+      - matchers
+  silences:
+    type: array
+    items:
+      $ref: '#/definitions/silence'
+  matchers:
+    type: array
+    items:
+      $ref: '#/definitions/matcher'
+    minItems: 1
+  matcher:
```

Comment:
How is a!="b" indicated?

Reply:
As far as I can tell as of today Alertmanager only supports equal or RegEx matching (See [internal matcher representation](https://github.com/prometheus/alertmanager/blob/master/types/match.go#L27)).

A user that would still like to input the semantics of a not-equal would need to use a RegEx.

Is this related to the API @brian-brazil?